### ISSUE TITLE
[TECHDEBT] [P2P] Raintree test suite improvements: Add configurations tests for dead and partially visible nodes

### ISSUE THREAD

**Issue by deblasis** [pokt-network/pocket#276]
## Objective

Tend to this TODO:

```go
./p2p/module_raintree_test.go:// TODO(olshansky): Add configurations tests for dead and partially visible nodes
```

Clarify scope and update below

## Origin Document

[ Why? Issue justification and/or link to another document]

## Goals

- _\_*REPLACE_ME*: List of things that are high level ideas or goals driving the task_
- [ ] ...

## Deliverable

- _\_*REPLACE_ME*: List of things that are concrete deliverables_
- [ ] ...

## Non-goals / Non-deliverables

- _\_*REPLACE_ME*: List of things that are out of scope_
- ...

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- \__REPLACE_ME_: Make sure to update the testing methodology appropriately\_
- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @deblasis 
**Co-Owners**: @Olshansk

**Comment by Olshansk** [comment]
This is blocked by #179, but once 179 is in, the goal is to update the testing suite to support the following test case:


<img width="800" alt="Screen Shot 2022-10-05 at 11 19 39 AM" src="https://user-images.githubusercontent.com/1892194/194133741-dec62423-1f4d-4e65-92c5-750197b7d067.png">

**Comment by Olshansk** [comment]
@andrewnguyen22 Is this something you're working on as part of RainTree cleanup & redundancy?
